tests/api/conftest.py

```diff
@@ -0,0 +1,79 @@
+import json
+
+import pytest
+
+from django.shortcuts import reverse
+from django.test.client import MULTIPART_CONTENT, Client
+from graphql_jwt.shortcuts import get_token
+
+from .utils import assert_no_permission
+
+API_PATH = reverse('api')
+
+
+class ApiClient(Client):
+    """GraphQL API client."""
+
+    def __init__(self, *args, **kwargs):
+        user = kwargs.pop('user')
+        self.user = user
+        self.token = get_token(user)
+        super().__init__(*args, **kwargs)
+
+    def _base_environ(self, **request):
+        environ = super()._base_environ(**request)
+        environ.update({'HTTP_AUTHORIZATION': 'JWT %s' % self.token})
+        return environ
+
+    def post(self, data=None, **kwargs):
+        """Send a POST request.
+
+        This wrapper sets the `application/json` content type which is
+        more suitable for standard GraphQL requests and doesn't mismatch with
+        handling multipart requests in Graphene.
+        """
+        if data:
+            data = json.dumps(data)
+        kwargs['content_type'] = 'application/json'
+        return super().post(API_PATH, data, **kwargs)
+
+    def post_graphql(
+            self, query=None, variables=None, permissions=None, **kwargs):
```

Comment:
How useful is the case where `query` is empty? I'd like to propose an alternative signature:

```python
def post_graphql(self, query, *, variables=None, permissions=None, **kwargs):
```

Reply:
Out of curiosity, why skip the single asterisk in the final implementation?

Reply:
Couldn't find a use for it in our current test structure, what's more, it enforced to pass the variables as the keyword adding more boilerplate.